plugins/bigquery/dbt/adapters/bigquery/impl.py

```diff
@@ -387,127 +388,31 @@ def load_dataframe(self, database, schema, table_name, agate_table,
-
-            'Clustering Fields',
-            clustering_value,
-            'The clustering fields for this table',
-            relation_type == 'table',
-        )
-        return zip(column_names, column_values)
-
     def get_catalog(self, manifest):
-        connection = self.connections.get_thread_connection()
-        client = connection.handle
-
-        schemas = manifest.get_used_schemas()
-
-        column_names = (
-            'table_database',
-            'table_schema',
-            'table_name',
-            'table_type',
-            'table_comment',
-            # does not exist in bigquery, but included for consistency
-            'table_owner',
-            'column_name',
-            'column_index',
-            'column_type',
-            'column_comment',
-        )
-        all_names = column_names + self._get_stats_column_names()
-        columns = []
+        """Get the catalog for this manifest by running the get catalog macro.
+        Returns an agate.Table of catalog information.
+        """
 
-        for database_name, schema_name in schemas:
-            relations = self.list_relations(database_name, schema_name)
-            for relation in relations:
-
-                # This relation contains a subset of the info we care about.
-                # Fetch the full table object here
-                table_ref = self.connections.table_ref(
-                    database_name,
-                    relation.schema,
-                    relation.identifier,
-                    connection
-                )
-                table = client.get_table(table_ref)
-
-                flattened = self._flat_columns_in_table(table)
-                relation_stats = dict(self._get_stats_columns(table,
-                                                              relation.type))
-
-                for index, column in enumerate(flattened, start=1):
-                    column_data = (
-                        relation.database,
-                        relation.schema,
-                        relation.name,
-                        relation.type,
-                        None,
-                        None,
-                        column.name,
-                        index,
-                        column.data_type,
-                        None,
-                    )
-                    column_dict = dict(zip(column_names, column_data))
-                    column_dict.update(copy.deepcopy(relation_stats))
-
-                    columns.append(column_dict)
-
-        return dbt.clients.agate_helper.table_from_data(columns, all_names)
+        information_schemas = []
+        for database, schema in manifest.get_used_schemas():
+            information_schemas.append(self.Relation.create(
+                database=database,
+                schema=schema,
+                quote_policy={
+                    'database': True,
+                    'schema': True
+                }
+            ))
```

Comment:
The existing `get_catalog`'s call to `_get_cache_schemas` should do this, and if it doesn't we should fix it for BigQuery! You might have to override the `BigQueryRelation.information_schema` method in some way.

Reply:
The big difference from the other `get_catalog` implementations here is that the BigQuery information schema is (usually) addressed with:
```
`project-id`.`dataset`.INFORMATION_SCHEMA.COLUMNS
```

ie. the information_schema is affixed to a dataset (schema) not a project (database). 

The exception is `SCHEMATA` which is addressed at the project level:
```
`project-id`.INFORMATION_SCHEMA.SCHEMATA
```

Do you still think we should override `get_cache_schemas`? I think I might also need to override `SchemaSearchMap` to return information schema Relations that BigQuery is happy with.

Reply:
Hmm, ok. In that case for now how about just implementing `_get_cache_schemas` as:
```
for database, schema in manifest.get_used_schemas():
    yield self.Relation.create(
        database=database,
        schema=schema,
        quote_policy={
            'database': True,
            'schema': True
        }
    )
```

In the long run, I think we should probably extend bigquery's Relation subclass to fully account for this quirky interpretation of `information_schema`, but we can do that at a later date.

Reply:
just a couple of great minds, thinking alike.